drivers/pinctrl/Kconfig.imx

```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2022 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+config PINCTRL_IMX
+	bool "Pin controller driver for iMX MCUs"
+	depends on HAS_MCUX_IOMUXC || HAS_IMX_IOMUXC
```

Comment:
why delete defaults based on dt compat?

Reply:
The `pinctrl_imx.c` driver can cover a large range of IOMUXC peripherals, all with different register layouts for their pad configuration register (which handles pullup/pulldown, open drain, etc). All of those peripherals of course require different DTS bindings. Therefore, I switched to using this approach to enabling the driver to avoid a large number of `default $(dt_compat_enabled,...)` strings here